### PR TITLE
pre_zokrates_pr

### PR DESCRIPTION
**Pull request by daniben31**

### CODE REVIEW

File: zokrates_core/Cargo.toml [hunk 1]
```diff
@@ -30,6 +30,10 @@ wasmi = "0.4.2"
 parity-wasm = "0.35.3"
 rustc-hex = "1.0"
 
+# Working on local version.
+zkinterface = { path = "../../zkinterface/rust" }
```
Comment (daniben31) [inline zokrates_core/Cargo.toml]: @naure Hi I think we need to write the url path and comment out the local path

File: zokrates_core/lib/cmake-build-debug/CMakeFiles/clion-log.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+CMakeLists.txt not found in /home/a/qed-it/ZoKrates/zokrates_core/lib Select CMakeLists.txt file...
```
Comment (daniben31) [inline zokrates_core/lib/cmake-build-debug/CMakeFiles/clion-log.txt]: @naure What is this?

File: zokrates_core/src/lib.rs [hunk 3]
```diff
@@ -19,6 +19,8 @@ extern crate serde_bytes;
 #[cfg(feature = "wasm")]
 extern crate wasmi;
 extern crate zokrates_field;
+extern crate zkinterface;
+//extern crate core;
```
Comment (daniben31) [inline zokrates_core/src/lib.rs]: @naure lets remove the comment

File: zokrates_core/lib/cmake-build-debug/CMakeFiles/clion-log.txt (hunk 2 above)
Comment (daniben31) [reply zokrates_core/lib/cmake-build-debug/CMakeFiles/clion-log.txt]: its generated by the ide, lets remove it

File: zokrates_core/src/component.rs [hunk 4]
```diff
@@ -0,0 +1,36 @@
+extern crate zkstandard;
```
Comment (daniben31) [inline zokrates_core/src/component.rs]: @naure we need to change the name of the file and its location

Comment (daniben31) [reply zokrates_core/src/component.rs]: i think that lib folder will fit better (location wise)

File: zokrates_core/src/proof_system/bn128/zkinterface.rs [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+
+    let args = R1CSConstraintsArgs { constraints: Some(vector_offset), info: None };
+
+    let r1cs_constraints = R1CSConstraints::create(&mut builder, &args);
+    let root_args = RootArgs { message_type: Message::R1CSConstraints, message: Some(r1cs_constraints.as_union_value()) };
+    let root = Root::create(&mut builder, &root_args);
+
+    builder.finish_size_prefixed(root, None);
+
+    println!("Writing {}", to_path);
```
Comment (daniben31) [inline zokrates_core/src/proof_system/bn128/zkinterface.rs]: @naure I didn't see print to stdout in other proof systems, should we keep this line anyway
?

File: zokrates_core/src/proof_system/bn128/zkinterface.rs [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+    let assign = Witness::create(&mut builder, &WitnessArgs {
+        assigned_variables: Some(values),
+    });
+    let message = Root::create(&mut builder, &RootArgs {
+        message_type: Message::Witness,
+        message: Some(assign.as_union_value()),
+    });
+    builder.finish_size_prefixed(message, None);
+
+    println!("Writing {}", to_path);
```
Comment (daniben31) [inline zokrates_core/src/proof_system/bn128/zkinterface.rs]: @naure same here

File: zokrates_core/src/proof_system/bn128/zkinterface.rs [hunk 7]
```diff
@@ -0,0 +1,229 @@
...
+        connections: VariablesOwned {
+            variable_ids: (0..first_local_id).collect(),
+            values,
+        },
+        free_variable_id,
+        r1cs_generation,
+        field_order: None,
+    };
+
+    println!("Writing {}", to_path);
```
Comment (daniben31) [inline zokrates_core/src/proof_system/bn128/zkinterface.rs]: @naure same here

**Review by daniben31 — COMMENTED** [review]
All comments are by @RonQedit 
We started the PR to be able to comment. Ron will approve

File: zokrates_core/src/component.rs (hunk 4 above)
Comment (naure) [reply zokrates_core/src/component.rs]: I deleted this file, it is not up-to-date.

File: zokrates_core/src/proof_system/bn128/zkinterface.rs (hunk 5 above)
Comment (naure) [reply zokrates_core/src/proof_system/bn128/zkinterface.rs]: True. On the other hand, ZoKrates writes human-readable outputs already and this helps a lot to understand that files were written.

**Pull request merged**